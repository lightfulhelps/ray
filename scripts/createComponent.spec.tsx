import { getNames, getComponentTemplate, getSpecTemplate } from './createComponent';

describe('getNames', () => {
  it('should throw if no argument', () => {
    expect(() => {
      getNames();
    }).toThrow('Invalid name');
  });

  it('should throw if empty string', () => {
    expect(() => {
      getNames('');
    }).toThrow('Invalid name');
  });

  it('should return an object containing componentName and className', () => {
    const names = getNames('foo');
    expect(names).toHaveProperty('componentName');
    expect(names).toHaveProperty('className');
  });

  it('should return componentName as start case with no space', () => {
    expect(getNames('foo').componentName).toEqual('Foo');
    expect(getNames('foo bar').componentName).toEqual('FooBar');
    expect(getNames('foo-bar').componentName).toEqual('FooBar');
    expect(getNames('foo bar foo').componentName).toEqual('FooBarFoo');
  });

  it('should return className as kebab case', () => {
    expect(getNames('foo').className).toEqual('foo');
    expect(getNames('foo bar').className).toEqual('foo-bar');
    expect(getNames('fooBar').className).toEqual('foo-bar');
    expect(getNames('foo bar foo').className).toEqual('foo-bar-foo');
  });
});

describe('getComponentTemplate', () => {
  it('matches snapshot', async () => {
    const names = { componentName: 'FooBar', className: 'foo-bar' };

    await expect(getComponentTemplate(names)).resolves.toMatchSnapshot();
  });
});

describe('getSpecTemplate', () => {
  it('matches snapshot', async () => {
    const names = { componentName: 'FooBar', className: 'foo-bar' };

    await expect(getSpecTemplate(names)).resolves.toMatchSnapshot();
  });
});
