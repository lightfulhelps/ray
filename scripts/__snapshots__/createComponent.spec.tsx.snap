// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentTemplate matches snapshot 1`] = `
"import * as React from 'react';
import classNames from 'classnames';

type Props = {
  className?: string;
  [key: string]: any;
};

const FooBar: React.FC<Props> = ({ className, ...other }) => {
   const classes = classNames(className, 'foo-bar');

   return <div {...other} className={classes} />;
};

export default FooBar;"
`;

exports[`getSpecTemplate matches snapshot 1`] = `
"import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import FooBar from './FooBar';

const setup = (overrides = {}) => {
  const props = merge({}, overrides);
  const wrapper = shallow(<FooBar {...props} />);

  return { wrapper, props };
};

describe('<FooBar />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('foo-bar')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
"
`;
